FinalProject/StateController.cpp

```diff
@@ -97,15 +96,22 @@ DWORD StateController::handleProtocolWriteEvents() {
 				else if (indexOfSignaledEvent == 4) {
 					setState(RTS);
 					releaseTX = !releaseTX;
+					break;
 					// set timeout to go to idle; also need to reset releaseTX to false when timeout fires
 				}
-				if (resentCounter++ == 3 && indexOfSignaledEvent == WAIT_TIMEOUT) {
+				if (indexOfSignaledEvent == WAIT_TIMEOUT) {
+					sendCommunicationMessageToCommController(9);
```

Comment:
Are we supposed to send an EOT when it times out?

Reply:
No, this is not correct. In fact, this logic was written to prevent the application from resending in  a loop. I think this should be using the errorCount and the breaks above should be handled better.